templates/pkg/resource/sdk.go.tpl

```diff
@@ -273,3 +273,55 @@ func (rm *resourceManager) terminalAWSError(err error) bool {
 	return false
 {{- end }}
 }
+
+{{- if .CRD.GetImmutableFields }}
+// validateImmutableFields validates the immutable fields and set appropriate condition
+func (rm *resourceManager) validateImmutableFields(ko *svcapitypes.{{ .CRD.Names.Camel }},
+	delta *ackcompare.Delta,) {
```

Comment:
Please pass a `*resource` pointer instead of `*svcapitypes.{{ .CRD.Names.Camel }}`, similar to the `updateConditions()`  method above, which has this function signature:

```go
func (rm *resourceManager) updateConditions (
        r *resource,
        err error,
) (*resource, bool) {
```

You can then get the `ko` object from `r` like so:

```go
        ko := r.ko
```

Also, please style the function signature like so:

```go
func (rm *resourceManager) validateImmutableFields(
        r *resource,
        delta *ackcompare.Delta,
) {
```